relay-general/src/pii/config.rs

```diff
@@ -199,23 +205,29 @@ pub struct RuleSpec {
 #[serde(rename_all = "camelCase")]
 pub struct Vars {
     /// The default secret key for hashing operations.
-    #[serde(default)]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
     pub hash_key: Option<String>,
 }
 
+impl Vars {
+    fn is_empty(&self) -> bool {
+        self.hash_key.is_none()
```

Comment:
Should we also consider post-trimmed empty strings as empty vars? In the tests below (from the diff), I only see `hash_key` being `null`.

Reply:
You might have more context than me here, when would `hash_key` be an empty string?